docs/docs/spec.md

```diff
@@ -281,8 +278,8 @@ Additionally, any location above is unrecognized if it makes up *any
 
 -   a local variable type
 -   an exception parameter
--   the type in a cast expression
--   an array or object creation expression
+-   the type in a cast or `instanceof` expression
+-   an array or object creation expression (including via a member reference)
```

Comment:
(I am getting myself confused as I try to differentiate parts of this bullet from the following bullet. But I'm going to say to just go ahead with this change: This whole section will be blown up and rewritten to accommodate our changing perspective toward implementation code, presumably following the structure of your Wiki page.)